### PR TITLE
reorder metadata legend before results

### LINKED ISSUE #180: `ml5.neuralNetwork` reorders labels that are numerical strings
Hi all,

I've been trying to train a [MNIST](https://yann.lecun.com/exdb/mnist/) handwritten digit classifier with `ml5.neuralNetwork` using numerical strings like `"1"`, `"2"`, `"3"` as labels. The trained neural network seems to classify inputs completely randomly even though the loss is converging.

After hours of debugging, I realized the neural network would behave as expected if I change my labels to be non-numerical strings like `"Digit 1"`, `"Digit 2"`, `"Digit 3"`. I did a few more [tests with the neuralNetwork color classifier example](https://editor.p5js.org/jackbdu/sketches/Y5nVeG8BX) and found out that numerical strings seem to be reordered when added as outputs, hence producing inaccurate classifications after training.

I haven't had a chance to look at the source code, but I suspect this might have something to do with how JavaScript treats numerical strings?

### PR DESCRIPTION
**Pull request by ziyuan-linn**
This PR fixes #180 by sorting `Object.entries(legend)` via the one hot array, ensuring that its ordering matches the ordering of `unformattedResults`.

### CODE REVIEW

**Comment by lindapaiste** [comment]
Can you explain what this sort function is doing?  It’s not clear at first glance.

It does feel like this is a roundabout solution.  It’s probably fine for now to just fix the issue but I wonder if there’s a better way to structure the metadata objects to avoid the issue.

**Comment by ziyuan-linn** [comment]
Thank you for pointing out the ambiguity. I also added a comment in the code.

This function reorders the `Object.entries(legend)` array from something like:
```js
[
  ["1", [0, 1, 0, 0],
  ["2", [1, 0, 0, 0],
  ["3", [0, 0, 0, 1],
  ["4", [0, 0, 1, 0],
]
```
to:
```js
[
  ["2", [1, 0, 0, 0],
  ["1", [0, 1, 0, 0],
  ["4", [0, 0, 1, 0], 
  ["3", [0, 0, 0, 1],
]
```

I agree that a better long-term solution is to restructure the metadata object. Perhaps we can track this in a new issue or add something to #112.

**Comment by shiffman** [comment]
@ziyuan-linn I'm going to go ahead and merge this for the "bug fix" but feel free to open an issue or add to an issue's discussion!

**Comment by lindapaiste** [comment]
I think you could get the same sort order with `vals.sort((a, b) => a[1].indexOf(1) - b[1].indexOf(1))`. Personally I find `.indexOf(1)` more clear than `.reduce((acc, curr) => acc * 2 + curr, 0)`, but this is already merged so no big deal.
